pcweb/pages/docs/custom_components.py

```diff
@@ -0,0 +1,218 @@
+import reflex as rx
+from pcweb.templates.docpage import docpage, h1_comp, text_comp
+import httpx
+import datetime
+
+# every app must have at least one tag in order to be rendered
+components_list = [
+    {
+        "package_name": "reflex-motion",
+        "tags": [],
+        "description": "A motion library for Reflex.",
+        "img": "https://reflexjs.org/static/preview.png",
+        "github": ""
+    },
+    {
+        "package_name": "reflex-chat",
+        "tags": [],
+        "description": "A chat component for Reflex.",
+        "img": "https://reflexjs.org/static/preview.png",
+        "github": ""
+    },
+    {
+        "package_name": "reflex-local-auth",
+        "tags": [],
+        "description": "Local authentication for Reflex.",
+        "img": "https://reflexjs.org/static/preview.png",
+        "github": ""
+    },
+    {
+        "package_name": "reflex-image-zoom",
+        "tags": [],
+        "description": "A zoomable image component for Reflex.",
+        "img": "https://reflexjs.org/static/preview.png",
+        "github": ""
+    },
+    {
+        "package_name": "reflex-webcam",
+        "tags": [],
+        "description": "A webcam component for Reflex.",
+        "img": "https://reflexjs.org/static/preview.png",
+        "github": ""
+    },
+    {
+        "package_name": "reflex-spline",
+        "tags": [],
+        "description": "A wrapper for the Spline design tool.",
+        "img": "https://reflexjs.org/static/preview.png",
+        "github": ""
+    },
+    {
+        "package_name": "reflex-color-picker",
+        "tags": [],
+        "description": "A color picker component for Reflex.",
+        "img": "https://reflexjs.org/static/preview.png",
+        "github": ""
+    },
+    {
+        "package_name": "reflex-simpleicons",
+        "tags": [],
+        "description": "SVG icons from the most popular brands.",
+        "img": "https://reflexjs.org/static/preview.png",
+        "github": ""
+    },
+
+]
+
+def fetch_url(url):
+    """Helper function to fetch data from a given URL."""
+    with httpx.Client() as client:
+        response = client.get(url)
+        if response.status_code == 200:
+            return response.json()
+        else:
+            print(f"Failed to fetch data from {url}: {response.status_code} - {response.text}")
+            return None
+        
+def get_downloads(package_name):
+    """Fetch the total downloads for the last month for a given package."""
```

Comment:
I think we should do the absolute total so the number seem bigger (for now, we can do monthly later). Alternatively I feel we can do last 30 days time instead of the previous month